app/client/websocket/index.ts

```diff
@@ -263,22 +247,24 @@ export default class WebSocketClient {
             } else if (this.eventCallback) {
                 if (reliableWebSockets) {
                     // We check the hello packet, which is always the first packet in a stream.
-                    if (msg.event === WebsocketEvents.HELLO && this.reconnectCallback) {
+                    if (msg.event === WebsocketEvents.HELLO) {
                         logInfo(this.url, 'got connection id ', msg.data.connection_id);
 
                         // If we already have a connectionId present, and server sends a different one,
                         // that means it's either a long timeout, or server restart, or sequence number is not found.
                         // If the server is not available the first time we try to connect, we won't have a connection id
                         // but still we need to sync.
                         // Then we do the sync calls, and reset sequence number to 0.
-                        if (
-                            (this.connectionId !== '' && this.connectionId !== msg.data.connection_id) ||
-                            (this.hasReliablyReconnect && this.connectionId === '')
-                        ) {
-                            logInfo(this.url, 'long timeout, or server restart, or sequence number is not found, or first connect after failure.');
-                            this.reconnectCallback();
+                        if (this.connectionId !== msg.data.connection_id) {
+                            if (this.connectionId) {
+                                logInfo(this.url, 'got a new connection due to long timeout, or server restart, or sequence number is not found');
+                            } else {
+                                logInfo(this.url, 'got the expected new connection id');
+                            }
+                            if (!this.shouldSkipSync) {
+                                this.reconnectCallback?.();
+                            }
```

Comment:
This means that the connection is new, so we should sync anyway

Reply:
The first time we connect we get through this point, because we have an empty connection id. It is a new connection, but the theory is that this is being handled outside of the websocket code (login or entry logic).